### PR TITLE
Add `Transformables` subscriber for improved `TransformContext` perf

### PR DESCRIPTION
**Pull request by teh-cmc**
Adds a `Transformables` subscriber that greatly improves `TransformContext`'s performance by preventing unnecessary queries (see comments in code for details).

Better name welcome.

- DNM: requires https://github.com/rerun-io/rerun/pull/6996

cc @Wumpf @jleibs 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6997?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6997?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/6997)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_space_view_spatial/src/contexts/transform_context.rs [hunk 1]
```diff
@@ -433,7 +433,18 @@ fn transform_at(
...
+    let is_potentially_transformed =
+        crate::transformables::Transformables::access(entity_db.store_id(), |transformables| {
+            transformables.is_potentially_transformed(entity_path)
+        })
+        .unwrap_or(false);
+    let transform3d = is_potentially_transformed
+        .then(|| get_parent_from_child_transform(entity_path, entity_db, query))
+        .flatten();
+
+    // let transform3d = get_parent_from_child_transform(entity_path, entity_db, query);
```
Comment (Wumpf) [inline crates/viewer/re_space_view_spatial/src/contexts/transform_context.rs:447]: leftover
```suggestion
```

File: crates/viewer/re_space_view_spatial/src/transformables.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+/// Keeps track of which entities have had any `Transform3D`-related data on any timeline at any
+/// point in time.
+///
+/// This is used to optimize queries in the `TransformContext`, so that we don't unnecessarily pay
+/// for the fixed overhead of all the query layers when we know for a fact that there won't be any
+/// data there.
+/// This is a huge performance improvement in practice, especially in recordings with many entities.
+#[derive(Default)]
+pub struct Transformables {
```
Comment (Wumpf) [inline crates/viewer/re_space_view_spatial/src/transformables.rs:22]: fancy choice of name. Wonder if that's what we could call these components user facing as well hum

**Review by Wumpf — APPROVED** [review]
👍

File: crates/viewer/re_space_view_spatial/src/transformables.rs (hunk 2 above)
Comment (Wumpf) [reply crates/viewer/re_space_view_spatial/src/transformables.rs:22]: how about `TransformComponentTracker` for this thing?

Comment (teh-cmc) [reply crates/viewer/re_space_view_spatial/src/transformables.rs:22]: Sure thing

**Pull request merged**